src/orquestra/sdk/_base/cli/_dorq/_login/_login.py

```diff
@@ -37,26 +42,67 @@ def __init__(
         self._config_repo: _repos.ConfigRepo = config_repo
         self._runtime_repo: _repos.RuntimeRepo = runtime_repo
         self._login_server: LoginServer = login_server
+        self._config_resolver: _arg_resolvers.ConfigResolver = config_resolver
 
-    def on_cmd_call(self, url: str, token: t.Optional[str], ce: bool):
+    def on_cmd_call(
+        self,
+        config: t.Optional[str],
+        url: t.Optional[str],
+        token: t.Optional[str],
+        ce: bool,
+    ):
+        """
+        Call the login command action, catching any exceptions that arise.
+        """
         try:
-            self._on_cmd_call_with_exceptions(url, token, ce)
+            self._on_cmd_call_with_exceptions(config, url, token, ce)
         except Exception as e:
             self._exception_presenter.show_error(e)
 
     def _on_cmd_call_with_exceptions(
         self,
-        url: str,
+        config: t.Optional[str],
+        url: t.Optional[str],
         token: t.Optional[str],
         ce: bool,
     ):
         """
         Implementation of the command action. Doesn't catch exceptions.
         """
+        assert bool(config) ^ bool(url), (
+            "orq login action was called with arguments "
+            f"'config = {config}, url = {url}'. "
+            "Exactly one of these arguments must be provided, but this constraint "
+            "should have been handled at CLI entry. Please report this as a bug."
+        )
+
+        _url: str
+        if config:
+            loaded_config = self._config_repo.read_config(
+                self._config_resolver.resolve(config)
+            )
+
+            if ce and loaded_config.runtime_name != RuntimeName.CE_REMOTE:
```

Comment:
So the inconsistency has its root in the fact that we have an explicit `--ce` flag but no `--qe` flag, so we have to make some kind of assumption when the flag isn't there. The assumption I've made is that when a user does `orq login -c my_ce_config` without `--ce` they want to log in to use CE. I think this makes sense since the whole point of login by config name is to streamline the process. (calling this the `lazy login assumption`) 

So, options:
a) CLI args are definitive - overwrite config contents
  - Pros - perfectly consistent
  - Cons - we don't have an explicit `QE` flag, so changing a config from CE to QE by accident is likely. Breaks lazy login assumption.
b) CLI args are definitive - check against config contents
  - Pros - perfectly consistent
  - Cons - Breaks lazy login assumption.
c) Config contents are definitive - check against CLI args (current implementation)
  - Pros - lazy login assumption
  - Cons - inconsistent due to the lack of a `QE` flag.
d) config contents are definitive - forbid `--ce` flag
  - Pros - lazy login assumption
  - Cons - None?

I personally favour d, but I'm happy to discuss the lazy login assumption if you disagree with it.

Reply:
Im more in favour of A) for 2 particular reasons:

1. I wouldnt assume that using -c or -s flag is anyhow different from each other. Now -c flag is <new> and is just a helper, but in the future I imagine it being the main go-to login method.  Thus I highly assume that using -c and -s flags on orq login should behave exactly the same. And now if I do login -s <> I login to QE. I assume doing login -c <> would login me to QE. With any other option than a), orq login -c and -s behave differently depending on what current confing stores. Plus its more logic - we dont like additional logic 😃 
2. There was a recent post that we will be adding --qe flag 😄 to make transition from QE to CE smoother.

Reply:
Good reasoning, do we have a ticket for the --qe flag so I can stick an appropriate todo in here?